src/mongocxx/collection.hpp

```diff
@@ -602,57 +562,51 @@ class MONGOCXX_API collection {
    private:
     friend class database;
 
-    MONGOCXX_PRIVATE collection(const database& database, bsoncxx::stdx::string_view collection_name);
+    MONGOCXX_PRIVATE collection(const database& database,
+                                bsoncxx::stdx::string_view collection_name);
+
+    MONGOCXX_PRIVATE collection(const database& database,
```

Comment:
Instead of creating the mongoc_colletion_t in `database.cpp`, maybe make this constructor take the options class and create it here? The db class could just pass the options along.